sklearn/neighbors/_base.py

```diff
@@ -469,20 +470,23 @@ def _fit(self, X, y=None):
             self._tree = X._tree
             self._fit_method = X._fit_method
             self.n_samples_fit_ = X.n_samples_fit_
+            self._n_features_out = self.n_samples_fit_
```

Comment:
`_n_features_out` is not going to be used by non-transformers. We should only set it in the transformers. For example:

https://github.com/scikit-learn/scikit-learn/blob/9daed7f8970c25afaa54f671815afd75605acf26/sklearn/neighbors/_graph.py#L376

Reply:
Done!

Reply:
Did you push the changes? it looks like `_n_features_out` is still defined in `NeighborsBase`

Reply:
This line should still be suppressed.

Reply:
Same comment for the other similar lines below in `NeighborsBase._fit`. They should be done only in `fit` / `_fit` methods of the subclasses that actually need it.

Reply:
Change is pushed! Had it stashed for some reason -- thanks for catching that!